src/Tyrannoport/assets/js/grid_filter.js

```diff
@@ -21,6 +32,19 @@ const initialiseFilter = element => {
         return true;
     }
 
+    const getUrl = () => {
+        const query = new URLSearchParams(window.location.search);
+        if (activeFilter.length > 0) {
+            query.set(targetSelector, activeFilter.join(","));
+        } else {
+            query.delete(targetSelector);
+        }
+        if (Array.from(query.keys()).length > 0) {
```

Comment:
if you wanted to avoid this, you could do something like:

```ts
const url = new URL(window.location.href);
if (activeFilter.length > 0) {
  url.searchParams.set(targetSelector, activeFilter.join(','));
} else {
  url.searchParams.delete(targetSelector);
}

return `${url.pathname}${url.search}`;
```